### ISSUE TITLE
Enable autostart when starting

### ISSUE THREAD

**Issue by woodcockjosh** [kubernetes/minikube#5301]
Consider adding a feature to enable autostart for minikube instances

For example: 
`minikube start -p hellokube --autostart=true`

This will autostart the instance when the host machine boots up / logs in. This will give a closer experience to how docker for desktop works. If additional configuration needs to be done fore each hypervisor then that is done as a one time setup the first time the configuration is used if it is not already done. For example, if this is on virtualbox then the feature will also auto configure virtualbox plist configuration.

**Comment by tstromberg** [comment]
Agreed. I would like to see this as part of our 2020 roadmap -- probably with a UI.

**Comment by priyawadhwa** [comment]
This feature is expected in 2020.

**Comment by brucelightyear** [comment]
Please keep this feature on track for 2020 :)

**Comment by priyawadhwa** [comment]
expected in 2020

**Comment by medyagh** [comment]
still on roadmap for 2020. once GUI lands

**Comment by tomekit** [comment]
Still on roadmap?

**Comment by brucelightyear** [comment]
I give up on this the bot close it...

**Comment by iamkarlson** [comment]
@priyawadhwa could you please reopen this issue as it's still very awaited?

**Comment by iamkarlson** [comment]
or @tstromberg

**Comment by brucelightyear** [comment]
Up !

**Comment by jufabeck2202** [comment]
Would love this feature for local dev.

**Comment by supanadit** [comment]
now its been 2022, and feature still expected

**Comment by iamkarlson** [comment]
any updates? still valuable

**Comment by PI-Victor** [comment]
/reopen
/remove-lifecycle rotten

still relevant

**Comment by brucelightyear** [comment]
Any news ?

**Comment by m18unet** [comment]
Still on roadmap?

**Comment by tstromberg** [comment]
To be honest, I feel like this feature is better handled by adding "minikube start" to your operating system's startup sequence. Adding built-in support for integrating with your operating systems scheduler feels very much like scope creep; particularly as it will need to handle 4 different scheduling systems (autostart, launchd, cron, systemd).

**Comment by woodcockjosh** [comment]
Docker starts automatically when I log in. I don't have to do something special for it. So does my google drive, and a host of other applications. @tstromberg What your suggesting is a workaround based on the lack of this feature. Just because it might seem hard to do with minikube's current architecture doesn't mean the developer wouldn't benefit from the feature.

**Comment by tstromberg** [comment]
Those are indeed very good points!

**Comment by iamkarlson** [comment]
still active
